abipy/flowtk/works.py

```diff
@@ -1967,3 +1968,156 @@ def on_all_ok(self):
+            omp_nbr_thread : Number of omp_thread to use.
+            manager: |TaskManager| of the task. If None, the manager is initialized from the config file.
+        """
+        # Verify phwork
+        if not isinstance(phwork, PhononWork):
+            raise TypeError("Work `%s` does not inherit from PhononWork" % phwork)
+        # Verify Multi
+        if (not withKerange) and (multi.ndtset != 4): #Without kerange, multi should contain 4 datasets
+            raise ValueError("""The |MultiDataset| object does not contain the expected number of dataset.
+                                It should have 4 datasets and it had `%s`. You should generate
+                                multi with the factory function conduc_from_scf_nscf_inputs""" % multi.ndtset)
+        if (withKerange) and (multi.ndtset != 6): # With Kerange, multi should contain 6 datasets
+            raise ValueError("""The |MultiDataset| object does not contain the expected number of dataset.
+                              It should have 6 datasets and it had `%s`. You should generate
+                              multi with the factory function conduc_from_scf_nscf_inputs""" % multi.ndtset)
+        # Verify nbr_proc and flow are defined if withKerange
+        if withKerange and (flow is None or nbr_proc is None):
+            raise ValueError("""When using kerange, the argument flow and nbr_proc must be passed to the function from_filepath
+                                flow = {}\n nbr_proc = {}""".format(flow, nbr_proc))
+
+        new = cls(manager=manager)
+
+        new.register_task(multi[0])
+        new.register_task(multi[1], deps={new[0]: "DEN"})
+        taskNumber = 2 # To keep track of the task in new and multi
+
+        if(withKerange): # Using Kerange
+            new.register_task(multi[2], deps={new[1]: "WFK"})
+            new.register_task(multi[3], deps={new[0]: "DEN", new[1]: "WFK", new[2]: "KERANGE.nc"})
+            taskNumber = 4 # We have 2 more dataset
+
+        if skipInter: # Without separate task for DVDB interpolation
+            new.register_task(multi[taskNumber+1], deps={new[taskNumber-1]: "WFK", phwork: ["DDB","DVDB"]})
+
+        else: # Separate task for DVDB interpolation
+            new.register_task(multi[taskNumber], deps={phwork: ["DDB","DVDB"]})
+            new.register_task(multi[taskNumber+1], deps={new[taskNumber-1]: "WFK", phwork: "DDB", new[taskNumber]: "DVDB"})
+
+        for task in new:
+            task.set_work(new)
+
+        # Manual Paralellization since autoparal doesn't work with optdriver=8 (t2)
+        if flow is not None :
+            new.set_flow(flow)
+            if nbr_proc is not None:
+                for task in new[2:]:
+                    task.with_fixed_mpi_omp(nbr_proc, omp_nbr_thread)
+        return new
+
+    @classmethod
+    def from_filepath(cls, DDB, DVDB, multi, nbr_proc=None, flow=None, withKerange=False,
```

Comment:
I would suggest to rename the variables using:

- DDB --> ddb_path
- DBDB --> dbdb_path

just to clarify that the method expects two strings with the file path and not a DdbFile object.